app/models/components/course/model_component_host.rb

```diff
@@ -0,0 +1,10 @@
+class Course::ModelComponentHost
+  include Componentize
+
+  Course.after_create do |course|
+    Course::ModelComponentHost.components.each do |component|
+      component.course_initialize(course)
```

Comment:
Maybe you should make the default module mixin define a default course_initialize, which does nothing. modules can override the method. See how I do it in the controller component host.